bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -737,6 +737,150 @@ public Image(Device device, ImageDataProvider imageDataProvider) {
 	init();
 }
 
+
+/**
+ * Upon zoom update, this method copies the image data and add disabled or grayed image depending on the flag
+ *
+ * @param device the device on which to create the image
+ * @param data the image data to create the image from (must not be null)
+ * @param flag the style, either <code>IMAGE_COPY</code>, <code>IMAGE_DISABLE</code> or <code>IMAGE_GRAY</code>
+ * @return new image data with valid styling properties as per the flag
+ */
+private ImageData copyImageDataForDisabledOrGrayImages(Device device, ImageData data, int flag) {
```

Comment:
Isn't that mostly duplicated what is in one of the Image constructor? I see there are some differences here and there, but it should be possible to extract the logic for IMAGE_DISABLE into one method and use it it both places and do the same for IMAGE_GRAY